hw/ip/otbn/doc/_index.md

```diff
@@ -426,6 +450,43 @@ These accesses are ignored if OTBN is busy.
 A host processor can check whether OTBN is busy by reading the {{< regref "STATUS.busy">}} flag.
 All memory accesses through the register interface must be word-aligned 32b word accesses.
 
+### Data Integrity Protection {#design-details-data-integrity-protection}
+
+#### Integrity Protection Code
+
+To provide overarching integrity protection for data without the need to regularly re-encode data OTBN uses the same integrity protection code everywhere.
+The code is applied to 32b data words, and produces 39b of encoded data.
+
+The used code is an inverted (39,32) [Hsiao](https://arxiv.org/pdf/0803.1217.pdf) SECDED error correction code.
+It has a minimum Hamming distance of four, resulting in the ability to detect at least three errors in a 32 bit word.
+The code is used for error detection only, no error correction is performed.
+
+In contrast to the [original Hsiao code](https://arxiv.org/pdf/0803.1217.pdf) the outputs are inverted to produce a code in which the all-zero word (0x0) is not a valid code word.
+
+#### Register File Integrity Protection
+
+OTBN contains two register files, the 32b GPRs, and the 256b WDRs.
+Both register files store error detection bits next to the actual data.
+
+The GPRs `x2` to `x31` store a 32b data word together with the integrity protection code, resulting in 39b of stored data.
+(The `x0`, the zero register, and `x1`, the call stack, require special treatment.)
+
+Each 256b wide data register (WDR) stores a 256b data word together with the integrity protection code, resulting in 312b of stored data.
+The integrity protection is done separately for each of the eight 32b sub-words within a 256b word.
+
+Both register files are able to accept protected data or unprotected data, and add integrity protection on demand before storing the data value.
```

Comment:
Hmm, I think that this needs slightly rejigging to make it clear that they only add check bits for unprotected data?